internal/utils/write_file.go

```diff
@@ -22,3 +24,33 @@ func WriteFile(path string, contents []byte, executable bool) error {
 
 	return nil
 }
+
+/* Writes a temp file and flushes the buffer */
+func CreateAndWriteTempFile(fileName string, data []byte) (string, error) {
+	f, err := os.CreateTemp("", fileName)
+	if err != nil {
+		return "", err
+	}
+	_, writeErr := f.Write(data)
+	if writeErr != nil {
+		return "", writeErr
+	}
+	f.Close()
```

Comment:
I was thinking we should close here so that we can call f.Sync in the next line and flush the buffer. If we defer f.Close() that would close the file at the end of the function right? Would the contents in memory still get flushed to disk?

Reply:
I believe `Sync()` can be called regardless of if the file is open or not.